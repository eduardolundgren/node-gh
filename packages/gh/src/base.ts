/**
 * Â© 2013 Liferay, Inc. <https://liferay.com> and Node GH contributors
 * (see file: CONTRIBUTORS)
 * SPDX-License-Identifier: BSD-3-Clause
 */

import Command, { flags } from '@oclif/command'
import * as fs from 'fs'

import * as configs from './configs'
import * as git from './git'

const config = configs.getConfig()
const testing = process.env.NODE_ENV === 'testing'

export default abstract class extends Command {
    public static flags: any = {
        help: flags.help({ char: 'h' }),
        debug: flags.boolean({
            description:
                'A more complete info flag, which leaks more privacy sensitive data by default.',
        }),
        info: flags.boolean({
            description: 'The info flag is useful for basic debugging',
        }),
        remote: flags.string({
            description: 'Override the default_remote setting in ~/.default.gh.json',
        }),
        repo: flags.string({ char: 'r', description: 'The repo to fetch issues from' }),
        user: flags.string({ char: 'u', description: 'The owner of the repository' }),
    }

    public flags
    public remoteInfo

    public async init() {
        this.setGlobalFlags()
        this.initLogger()
    }

    private initLogger() {
        process.env.DEBUG = this.flags.debug
        process.env.INFO = this.flags.info
    }

    private setGlobalFlags() {
        // @ts-ignore: need to figure out if this error is benign
        const { flags } = this.parse(this.constructor)

        flags.remote = flags.remote || config.default_remote

        const remoteUrl = git.getRemoteUrl(flags.remote)

        flags.isTTY = {}
        flags.isTTY.in = Boolean(process.stdin.isTTY)
        flags.isTTY.out = Boolean(process.stdout.isTTY)
        flags.loggedUser = getUser()
        flags.remoteUser = git.getUserFromRemoteUrl(remoteUrl)

        if (!flags.user) {
            if (flags.repo || flags.all) {
                flags.user = flags.loggedUser
            } else {
                flags.user = process.env.GH_USER || flags.remoteUser || flags.loggedUser
            }
        }

        flags.repo = flags.repo || git.getRepoFromRemoteURL(remoteUrl)
        flags.currentBranch = testing ? 'master' : git.getCurrentBranch()
        flags.github_host = config.github_host
        flags.github_gist_host = config.github_gist_host

        this.flags = flags
    }
}

const verbose = process.argv.indexOf('--verbose') !== -1
const insane = process.argv.indexOf('--insane') !== -1

process.on('unhandledRejection', r => console.log(r))

if (verbose || insane) {
    process.env.GH_VERBOSE = 'true'
}

if (insane) {
    process.env.GH_VERBOSE_INSANE = 'true'
}

// -- Config -------------------------------------------------------------------

export function clone(o) {
    return JSON.parse(JSON.stringify(o))
}

// -- Utils --------------------------------------------------------------------

export function load() {}

export function find(filepath, opt_pattern) {
    return fs.readdirSync(filepath).filter(file => {
        return (opt_pattern || /.*/).test(file)
    })
}

export function getUser() {
    return config.github_user || process.env.GH_USER
}

// Export some config methods to allow plugins to access them
export const getConfig = configs.getConfig
export const writeGlobalConfig = configs.writeGlobalConfig
