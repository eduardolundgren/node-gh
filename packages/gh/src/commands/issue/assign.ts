import Command from '../../base'
import { afterHooks, beforeHooks } from '../../hooks'
import * as logger from '../../logger'
import { getUserRepo } from '../../utils'
import { editIssue, getIssue } from './index'

export default class Assign extends Command {
    static aliases = ['i:a', 'issue:a', 'issue:assign']

    public static description = 'Assign issue to user'

    public static args = [
        {
            name: 'number',
            required: true,
            description: 'Number(s) of the issue.',
            parse: input => input.split(','),
        },
        {
            name: 'assignee',
            required: true,
            description: 'User to assign the issue to.',
        },
    ]

    public static flags = {
        ...Command.flags,
    }

    public async run() {
        const { args } = this.parse(Assign)

        await runAssignCmd({ ...this.flags, ...args }).catch(err =>
            logger.error(err, 'Error assigning issue')
        )
    }
}

export async function runAssignCmd(flags) {
    await beforeHooks('issue.assign', flags)

    for (const num of flags.number) {
        logger.log(
            `Assigning issue ${num} on ${getUserRepo(flags)} to ${logger.colors.magenta(
                flags.assignee
            )}`
        )

        const issue = await getIssue(num, flags)

        const { data } = await editIssue(num, issue.title, 'open', flags)

        logger.log(logger.colors.cyan(data.html_url))
    }

    await afterHooks('issue.assign', flags)
}
