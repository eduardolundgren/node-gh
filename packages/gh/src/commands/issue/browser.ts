import * as openUrl from 'opn'
import Command from '../../base'
import { afterHooks, beforeHooks } from '../../hooks'
import * as logger from '../../logger'

export default class Browser extends Command {
    public static aliases = ['i:b', 'issue:b', 'issue:browser']

    public static args = [
        {
            name: 'number',
            required: true,
            description: 'Number(s) of the issue.',
            parse: input => input.split(','),
        },
    ]

    public static description = 'Browser & filter issues'

    public static flags = {
        ...Command.flags,
    }

    public async run() {
        const { args } = this.parse(Browser)

        await runBrowserCmd({ ...this.flags, ...args }).catch(err =>
            logger.error(err, 'Error running browser cmd.')
        )
    }
}

export async function runBrowserCmd(flags) {
    await beforeHooks('issue.browser', flags)

    for (const num of flags.number) {
        openUrl(`${flags.github_host}/${flags.user}/${flags.repo}/issues/${num}`, {
            wait: false,
        })
    }

    await afterHooks('issue.browser', flags)
}
