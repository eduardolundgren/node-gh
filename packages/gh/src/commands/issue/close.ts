import Command from '../../base'
import { afterHooks, beforeHooks } from '../../hooks'
import * as logger from '../../logger'
import { getUserRepo } from '../../utils'
import { editIssue, getIssue } from './index'

export default class Close extends Command {
    static aliases = ['i:c', 'issue:c', 'issue:close']

    public static args = [
        {
            name: 'number',
            required: true,
            description: 'Number(s) of the issue.',
            parse: input => input.split(','),
        },
    ]

    public static description = 'Close Issue'

    public static flags = {
        ...Command.flags,
    }

    public async run() {
        const { args } = this.parse(Close)

        runCloseCmd({ ...this.flags, ...args }).catch(err => logger.error(`Can't close issue.`))
    }
}

export async function runCloseCmd(flags) {
    await beforeHooks('issue.close', flags)

    for (const number of flags.number) {
        logger.log(`Closing issue ${number} on ${getUserRepo(flags)}`)

        const issue = await getIssue(number, flags)
        const { data } = await editIssue(number, issue.title, 'closed', flags)

        logger.log(logger.colors.cyan(data.html_url))
    }

    await afterHooks('issue.close', flags)
}
