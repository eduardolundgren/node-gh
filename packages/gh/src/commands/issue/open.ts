import Command from '../../base'
import { afterHooks, beforeHooks } from '../../hooks'
import * as logger from '../../logger'
import { getUserRepo } from '../../utils'
import { editIssue, getIssue } from './index'

export default class Open extends Command {
    static aliases = ['i:o', 'issue:o', 'issue:open']

    public static args = [
        {
            name: 'number',
            required: true,
            description: 'Number(s) of the issue.',
            parse: input => input.split(','),
        },
    ]

    public static description = 'Open Issue'

    public static flags = {
        ...Command.flags,
    }

    public async run() {
        const { args } = this.parse(Open)

        await runOpenCmd({ ...this.flags, ...args }).catch(err => logger.error("Can't open issue."))
    }
}

export async function runOpenCmd(flags) {
    await beforeHooks('issue.open', flags)

    for (const number of flags.number) {
        logger.log(`Opening issue ${number} on ${getUserRepo(flags)}`)

        const issue = await getIssue(number, flags)
        const { data } = await editIssue(number, issue.title, 'open', flags)

        logger.log(logger.colors.cyan(data.html_url))
    }

    await afterHooks('issue.open', flags)
}
