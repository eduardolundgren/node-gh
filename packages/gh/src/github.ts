/**
 * Â© 2013 Liferay, Inc. <https://liferay.com> and Node GH contributors
 * (see file: CONTRIBUTORS)
 * SPDX-License-Identifier: BSD-3-Clause
 */

import * as Octokit from '@octokit/rest'
import * as level from 'console-log-level'
import * as fs from 'fs'
import * as inquirer from 'inquirer'
import { get } from 'lodash'
import * as moment from 'moment'
import { join } from 'path'
import * as userhome from 'userhome'
import { getConfig, writeGlobalConfigCredentials } from './configs'
import * as logger from './logger'

const config = getConfig()

let githubAPI

export async function github(apiPath: string, payload): Promise<Octokit.AnyResponse | any> {
    if (!githubAPI) {
        githubAPI = await getGitHubInstance()
    }

    if (apiPath.includes('list')) {
        const githubMethod = get(githubAPI, apiPath).endpoint(payload)

        return githubAPI
            .paginate(githubMethod)
            .catch(err => logger.error(err, 'Error when calling paginated GitHub API endpoint'))
    }

    return get(githubAPI, apiPath)(payload).catch(err =>
        logger.error(err, 'Error when calling GitHub API endpoint')
    )
}

export async function getGitHubInstance(): Promise<Octokit> {
    const baseUrl =
        config.github_host === 'https://github.com' ? 'https://api.github.com' : config.github_host
    const token = await getToken()

    const throttlePlugin = await import('@octokit/plugin-throttling')

    Octokit.plugin(throttlePlugin)

    let logLevel = 'error'

    if (process.argv.includes('--debug')) {
        logLevel = 'debug'
    } else if (process.argv.includes('--info')) {
        logLevel = 'info'
    }

    return new Octokit({
        baseUrl,
        auth: token,
        log: level({
            level: logLevel,
        }),
        throttle: {
            onRateLimit: (retryAfter, options) => {
                console.warn(`Request quota exhausted for request ${options.method} ${options.url}`)

                if (options.request.retryCount === 0) {
                    // only retries once
                    console.log(`Retrying after ${retryAfter} seconds!`)
                    return true
                }
            },
            onAbuseLimit: (_, options) => {
                // does not retry, only logs a warning
                console.warn(`Abuse detected for request ${options.method} ${options.url}`)
            },
        },
    })
}

export async function getToken(): Promise<string> {
    let token

    if (!tokenExists()) {
        token = await createNewOathToken()
    } else {
        token = getSavedToken()
    }

    if (token) {
        return token
    }

    process.exit(1)
}

export function tokenExists(): boolean {
    if (process.env.GENERATE_NEW_TOKEN) {
        return false
    }

    return (
        (config.github_token && config.github_user) || (process.env.GH_TOKEN && process.env.GH_USER)
    )
}

function getSavedToken(): string {
    if (process.env.CONTINUOUS_INTEGRATION) {
        return process.env.GH_TOKEN
    }
    if (process.env.NODE_ENV === 'testing') {
        // Load your local token when generating test fixtures
        return JSON.parse(fs.readFileSync(userhome('.gh.json')).toString()).github_token
    }

    return config.github_token
}

async function createNewOathToken(): Promise<string | undefined> {
    logger.log(`In order to use GitHub's API, you will need to login with your GitHub account.`)

    interface Answers {
        username: string
        password: string
    }

    const answers: Answers = await inquirer.prompt([
        {
            type: 'input',
            message: 'Enter your GitHub user',
            name: 'username',
        },
        {
            type: 'password',
            message: 'Enter your GitHub password',
            name: 'password',
        },
    ])

    const octokit = new Octokit({
        auth: {
            username: answers.username,
            password: answers.password,
            async on2fa() {
                const { code } = await inquirer.prompt([
                    {
                        type: 'input',
                        message: 'Enter your Two-factor GitHub authenticator code',
                        name: 'code',
                    },
                ])

                return code
            },
        },
    })

    const payload = {
        note: `Node GH (${moment().format('MMMM Do YYYY, h:mm:ss a')})`,
        note_url: 'https://github.com/node-gh/gh',
        scopes: ['user', 'public_repo', 'repo', 'repo:status', 'delete_repo', 'gist'],
    }

    try {
        var { data } = await octokit.oauthAuthorizations.createAuthorization(payload)
    } catch (err) {
        throw new Error(`Error creating GitHub token\n${err.message}`)
    }

    if (data.token) {
        writeGlobalConfigCredentials(
            answers.username,
            data.token,
            process.env.GENERATE_NEW_TOKEN && join(__dirname, '../__tests__/auth.json')
        )

        return data.token
    }

    logger.log(`Was not able to retrieve token from GitHub's api`)
}
