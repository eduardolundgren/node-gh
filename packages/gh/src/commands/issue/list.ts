import { flags } from '@oclif/command'
import { isArray } from 'lodash'
import Command from '../../base'
import { github } from '../../github'
import { afterHooks, beforeHooks } from '../../hooks'
import * as logger from '../../logger'
import { getUserRepo } from '../../utils'

export const listCmdFlags = {
    all: flags.boolean({ char: 'a', description: 'List all issues' }),
    assignee: flags.string({
        char: 'A',
        description: 'Filter issues by assignee(case sensitive) login id',
    }),
    detailed: flags.boolean({ char: 'd', description: 'Show detailed version of issues' }),
    label: flags.string({
        char: 'L',
        description: 'Filter issues by label(s). If multiple labels they should be comma separated',
    }),
    milestone: flags.string({
        char: 'M',
        description: 'Filter issues by milestone (case insensitive)',
    }),
    state: flags.string({
        required: false,
        default: 'open',
        description: 'Filter issues by state',
        options: ['open', 'closed'],
    }),
}

export default class List extends Command {
    static aliases = ['i:l', 'issue:l', 'issue:list']

    public static description = 'List & filter issues'

    public static flags = {
        ...Command.flags,
        ...listCmdFlags,
    }

    public async run() {
        runListCmd(this.flags).catch(err => logger.error(err, 'Error listing issues.'))
    }
}

export async function runListCmd(flags) {
    beforeHooks('issue.list', flags)

    if (flags.all) {
        logger.log(
            `Listing ${logger.colors.green(flags.state)} issues for ${logger.colors.green(
                flags.user
            )}`
        )

        await listFromAllRepositories(flags.user)
    } else {
        logger.log(`Listing ${logger.colors.green(flags.state)} issues on ${getUserRepo(flags)}`)

        await list(flags)
    }

    afterHooks('issue.list', flags)
}

async function list(flags, repo?: string) {
    let payload

    payload = {
        repo: repo || flags.repo,
        owner: flags.user,
        state: flags.state,
    }

    if (flags.labels) {
        payload.labels = flags.labels
    }

    if (flags['no-milestone']) {
        payload.milestone = 'none'
    }

    if (flags.milestone) {
        const data = await github('issues.listMilestonesForRepo', {
            repo: flags.repo,
            owner: flags.user,
        })

        const milestoneNumber = data.filter(byMilestoneTitle).map(getNumber)

        payload.milestone = `${milestoneNumber}`
    }

    function byMilestoneTitle(milestone) {
        return flags.milestone === milestone.title
    }

    function getNumber(milestone) {
        return milestone.number
    }

    if (flags.assignee) {
        payload.assignee = flags.assignee
    }

    const data = await github('issues.listForRepo', payload)

    const issues = data.filter(result => Boolean(result))

    if (issues && issues.length > 0) {
        const formattedIssues = formatIssues(issues, flags.detailed)

        flags.all
            ? logger.log(`\n${getUserRepo(this)}:\n${formattedIssues}`)
            : logger.log(formattedIssues)
    } else {
        logger.log(`\nNo issues on ${getUserRepo(flags)}`)
    }
}

async function listFromAllRepositories(user) {
    const payload = {
        type: 'all',
        username: user,
    }

    const repositories: any = await github('repos.listForUser', payload)

    for (const repo of repositories) {
        await list(repo.owner.login, repo.name)
    }
}

function formatIssues(issues, showDetailed, dateFormatter?: string) {
    issues.sort((a, b) => {
        return a.number > b.number ? -1 : 1
    })

    if (issues && issues.length > 0) {
        const formattedIssuesArr = issues.map(issue => {
            const issueNumber = logger.colors.green(`#${issue.number}`)
            const issueUser = logger.colors.magenta(`@${issue.user.login}`)
            const issueDate = `(${logger.getDuration(issue.created_at, dateFormatter)})`

            let formattedIssue = `${issueNumber} ${issue.title} ${issueUser} ${issueDate}`

            if (showDetailed) {
                if (issue.body) {
                    formattedIssue = `
                        ${formattedIssue}
                        ${issue.body}
                    `
                }

                if (isArray(issue.labels) && issue.labels.length > 0) {
                    const labels = issue.labels.map(label => label.name)
                    const labelHeading = labels.length > 1 ? 'labels: ' : 'label: '

                    formattedIssue = `
                        ${formattedIssue}
                        ${logger.colors.yellow(labelHeading) + labels.join(', ')}
                    `
                }

                if (issue.milestone) {
                    const { number, title } = issue.milestone

                    formattedIssue = `
                        ${formattedIssue}
                        ${`${logger.colors.green('milestone: ')} ${title} - ${number}`}
                    `
                }

                formattedIssue = `
                    ${formattedIssue}
                    ${logger.colors.blue(issue.html_url)}
                `
            }

            return trim(formattedIssue)
        })

        return formattedIssuesArr.join('\n\n')
    }

    return null
}

function trim(str) {
    return str
        .replace(/^[ ]+/gm, '')
        .replace(/[\r\n]+/g, '\n')
        .trim()
}
