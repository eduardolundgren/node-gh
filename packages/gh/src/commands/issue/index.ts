import { forEach } from 'lodash'
import Command from '../../base'
import { IFlags } from '../../interfaces'
// import { runCommentCmd } from './comment'
// import { runBrowserCmd } from './browser'
// import { newCmdFlags, runNewCmd } from './new'
import { listCmdFlags, runListCmd } from './list'

export default class Issue extends Command {
    static aliases = ['is']

    public static args = [
        {
            name: 'number_or_title',
            required: false,
            description:
                'Number of the issue you would like to target OR the text message of a new issue',
        },
        {
            name: 'comment_or_body',
            required: false,
            description:
                'Comment message you would like to add to the issue OR the body message of a new issue',
        },
    ]

    public static description = 'List, Create & Modify issues'

    public static flags = generateFlags()

    public async run() {
        const { args } = this.parse(Issue)
        const flags = this.flags

        // const isShortcutForBrowser = Number(args.number_or_title) && !args.comment_or_body
        // const isShortcutForComment = Number(args.number_or_title) && args.comment_or_body
        const isShorcutForListIssues = !args.number_or_title && !args.comment_or_body
        // const isShorcutForNewIssue = !Number(args.number_or_title) && args.comment_or_body

        // if (isShortcutForBrowser) {
        //   runBrowserCmd(args.number_or_title, this.remoteInfo)
        // } else if (isShortcutForComment) {
        //   const adjustedArgs = {
        //     number: args.number_or_title,
        //     message: args.comment_or_body
        //   }

        //   runCommentCmd(adjustedArgs, this.remoteInfo)
        // } else
        if (isShorcutForListIssues) {
            runListCmd(flags)
        }
        // else if (isShorcutForNewIssue) {
        //   const adjustedFlags = {
        //     ...flags,
        //     title: args.number_or_title,
        //     message: args.comment_or_body,
        //   }

        //   runNewCmd(adjustedFlags, this.remoteInfo)
        // }
    }
}

function generateFlags() {
    const flagsArr = [
        { flags: { ...Command.flags }, namespace: 'global' },
        { flags: { ...listCmdFlags }, namespace: 'issue:list' },
        // { flags: { ...newCmdFlags }, namespace: 'issue:new' },
    ]

    return flagsArr.reduce((previousFlags, currentFlags) => {
        const { flags } = currentFlags

        let allFlags = { ...previousFlags }

        forEach(flags, (flag: IFlags, key) => {
            let description = `\`${currentFlags.namespace}\` --> ${flag.description}`

            if (previousFlags.hasOwnProperty(key)) {
                description = `${previousFlags[key].description}\n ${description}`
            }

            allFlags[key] = {
                ...flag,
                description,
            }
        })

        return allFlags
    }, {})
}
