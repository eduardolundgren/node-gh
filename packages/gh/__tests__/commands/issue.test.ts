/**
 * Â© 2013 Liferay, Inc. <https://liferay.com> and Node GH contributors
 * (see file: CONTRIBUTORS)
 * SPDX-License-Identifier: BSD-3-Clause
 */

import * as opn from 'opn'
import { default as Browser, default as List } from '../../src/commands/issue'
import Assign from '../../src/commands/issue/assign'
import Close from '../../src/commands/issue/close'
import Open from '../../src/commands/issue/open'
import { prepareTestFixtures } from '../../src/utils'
import { runCmd } from '../runCmd'

let spy

beforeEach(() => {
    spy = jest.spyOn(global.console, 'log')
})

afterEach(() => {
    spy.mockRestore()
})

describe('E2E: Issues Module Test', () => {
    it('List Issues Shorthand', async () => {
        const { nockDone } = await prepareTestFixtures('issue.list')

        await List.run([])

        expect(spy.mock.calls).toMatchSnapshot()

        nockDone()
    })

    it('Assign Issues', async () => {
        const { nockDone } = await prepareTestFixtures('issue.assign')

        await Assign.run(['1', process.env.GH_USER])

        expect(spy.mock.calls).toMatchSnapshot()

        nockDone()
    })

    it('Open Issue in Browser', async () => {
        await Browser.run(['7,10'])

        expect(opn).toHaveBeenCalledTimes(2)
        expect(opn).toHaveBeenCalledWith('https://github.com/protoEvangelion/gh/issues/7', {
            wait: false,
        })
    })

    xit('Create a new issue `gh is -N -t "Node GH rocks!" -L bug,question,test`', done => {
        expect(runCmd('gh is -N -t "Node GH rocks!" -L bug,question,test')).toMatchSnapshot()
        done()
    })

    xit('Comment on Issues `gh is 1 -c "comment"`', done => {
        expect(runCmd('gh is 1 -c "comment"')).toMatchSnapshot()
        done()
    })

    it('Close Issue', async () => {
        const { nockDone } = await prepareTestFixtures('issue.close')

        await Close.run(['7,10'])

        expect(spy.mock.calls).toMatchSnapshot()

        nockDone()
    })

    it('Open Issue', async () => {
        const { nockDone } = await prepareTestFixtures('issue.open')

        await Open.run(['7,10'])

        expect(spy.mock.calls).toMatchSnapshot()

        nockDone()
    })

    xit('Search Issues `gh is -s hi`', done => {
        expect(runCmd("gh is -s 'hi'")).toMatchSnapshot()
        done()
    })
})
