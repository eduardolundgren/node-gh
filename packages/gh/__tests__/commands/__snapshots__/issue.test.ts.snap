// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`E2E: Issues Module Test Assign Issues 1`] = `
Array [
  Array [
    "Assigning issue 1 on protoEvangelion/gh to protoEvangelion",
  ],
  Array [
    "https://github.com/protoEvangelion/gh/issues/1",
  ],
]
`;

exports[`E2E: Issues Module Test Close Issue 1`] = `
Array [
  Array [
    "Closing issue 7 on protoEvangelion/gh",
  ],
  Array [
    "https://github.com/protoEvangelion/gh/issues/7",
  ],
  Array [
    "Closing issue 10 on protoEvangelion/gh",
  ],
  Array [
    "https://github.com/protoEvangelion/gh/issues/10",
  ],
]
`;

exports[`E2E: Issues Module Test Comment on Issues \`gh is 1 -c "comment"\` 1`] = `
"Adding comment on issue #1 on protoEvangelion/gh
https://github.com/protoEvangelion/gh/issues/1#issuecomment-000000001
"
`;

exports[`E2E: Issues Module Test Create a new issue \`gh is -N -t "Node GH rocks!" -L bug,question,test\` 1`] = `
"Creating a new issue on protoEvangelion/gh
https://github.com/protoEvangelion/gh/issues/88
{after-hook} gh is --browser  --user protoEvangelion --repo gh --number 88
"
`;

exports[`E2E: Issues Module Test List Issues Shorthand 1`] = `
Array [
  Array [
    "Listing open issues on protoEvangelion/gh",
  ],
  Array [
    "#99 notification @protoEvangelion (in 4 years)

#98 pr title2 @protoEvangelion (in 4 years)

#91 Node GH rocks! @protoEvangelion (in 4 years)

#42 test2 @protoEvangelion (in 4 years)

#40 My pr title @protoEvangelion (in 4 years)

#38 Node GH rocks! @protoEvangelion (in 4 years)",
  ],
]
`;

exports[`E2E: Issues Module Test Open Issue 1`] = `
Array [
  Array [
    "Opening issue 7 on protoEvangelion/gh",
  ],
  Array [
    "https://github.com/protoEvangelion/gh/issues/7",
  ],
  Array [
    "Opening issue 10 on protoEvangelion/gh",
  ],
  Array [
    "https://github.com/protoEvangelion/gh/issues/10",
  ],
]
`;

exports[`E2E: Issues Module Test Search Issues \`gh is -s hi\` 1`] = `
"Searching for hi in issues for protoEvangelion/gh

#72 hi @protoEvangelion (in 4 years)

#65 hi @protoEvangelion (in 4 years)

#63 hi @protoEvangelion (in 4 years)

#62 hi @protoEvangelion (in 4 years)

#61 hi @protoEvangelion (in 4 years)

#60 hi @protoEvangelion (in 4 years)

#19 hi @protoEvangelion (in 4 years)

#1 test1 @protoEvangelion (in 4 years)
"
`;
